experimental/docker-build-with-squash.md

```diff
@@ -0,0 +1,49 @@
+# Docker build --squash
+
+`--squash` is a experimental CLI argument to docker build that allows your built images to be squashed to their parent, which does not destory any images or layers, and preserves the build cache.
+
+Once the build is complete, docker creates a new image loading the diffs from each layer into a single new layer and references all the parent's layers.
+
+## The following is an example of docker build with `--squash` argument
+
+```Dockerfile
+
+FROM busybox
+RUN echo hello > /hello
+RUN echo world >> /hello
+RUN touch remove_me /remove_me
+ENV HELLO world
+RUN rm /remove_me
+
+```
+An image named `test` is built with `--squash`.
+
+```bash
+$ docker build -t test squash .
+
+[...]
+
+```
+
+If everything is right, the history will look like this:
+
+```
+$ docker history test 
+
+IMAGE               CREATED             CREATED BY                                      SIZE                COMMENT
+4e10cb5b4cac        3 seconds ago                                                       12 B                merge sha256:88a7b0112a41826885df0e7072698006ee8f621c6ab99fca7fe9151d7b599702 to sha256:47bcc53f74dc94b1920f0b34f6036096526296767650f223433fe65c35f149eb
+<missing>           5 minutes ago       /bin/sh -c rm /remove_me                        0 B
+<missing>           5 minutes ago       /bin/sh -c #(nop) ENV HELLO=world               0 B
+<missing>           5 minutes ago       /bin/sh -c touch remove_me /remove_me           0 B
+<missing>           5 minutes ago       /bin/sh -c echo world >> /hello                 0 B
+<missing>           6 minutes ago       /bin/sh -c echo hello > /hello                  0 B
+<missing>           7 weeks ago         /bin/sh -c #(nop) CMD ["sh"]                    0 B
+<missing>           7 weeks ago         /bin/sh -c #(nop) ADD file:47ca6e777c36a4cfff   1.113 MB
+
+```
+We could find that all layer's name is <missing>, and there is a new layer with COMMENT `merge`.
```

Comment:
I think we cannot using <missing> here, since the markdown format will make it missing.